File: src/main/java/com/team/puddy/domain/user/repository/UserRepository.java
package com.team.puddy.domain.user.repository;

import com.team.puddy.domain.type.JwtProvider;
import com.team.puddy.domain.user.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

import javax.swing.text.html.Option;
import java.util.Optional;

public interface UserRepository extends JpaRepository<User,Long>, UserQueryRepository {

    Optional<User> findByAccount(String account);

    Optional<User> findByUsernameAndEmail(String username, String email);

    Optional<User> findByAccountAndUsernameAndEmail(String account, String username, String email);


    Optional<User> findByAccountAndEmail(String sub, String email);
}

File: src/main/kotlin/com/team/puddy/domain/user/repository/UserRepository.kt
package com.team.puddy.domain.user.repository;

import com.team.puddy.domain.user.domain.User
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

interface UserRepository : JpaRepository<User, Long> {

    fun findByAccount(account: String): Optional<User>

    fun findByUsernameAndEmail(username: String, email: String): Optional<User>

    fun findByAccountAndUsernameAndEmail(account: String, username: String, email: String): Optional<User>

    fun findByAccountAndEmail(sub: String, email: String): Optional<User>

    fun existsByEmail(email: String): Boolean
}
